src/fetch.py

```diff
@@ -89,6 +99,97 @@ def read_json(self, file):
         except IOError:
             return False
 
+    def get_server_version(self):
+        """Get version of server which is being processed
+
+        Note that the path to the version is the same across Ping Federate
+        versions."""
+        api_path = f'{self.ping_data["basePath"]}/version'
+        response = self.session.get(urljoin(self.swagger_url, api_path))
+        response.raise_for_status()
+        response = response.json()
+        return response['version']
+
+    def apply_20_patches(self, patch_root: Path):
+        self.logger.debug('Searching for patches in:%s', patch_root)
+        if not patch_root.exists():
```

Comment:
Are we making the assumption that the patch root should always exist, and it not existing is an error? I would imagine that there are instances where the patches are not necessary, for-instance, if the user is running the latest version of the products?